hypersync-client/src/lib.rs

```diff
@@ -192,20 +214,22 @@ impl Client {
 
     /// Internal constructor that takes both config and user agent.
     fn new_internal(cfg: ClientConfig, user_agent: String) -> Result<Self> {
-        let http_client = reqwest::Client::builder()
-            .no_gzip()
-            .timeout(Duration::from_millis(cfg.http_req_timeout_millis))
```

Comment:
Instead of adding timeout to the client, I now add it on each `.request` (which the client was doing internally)